drivers/tee/optee/optee_msg.h

```diff
@@ -429,6 +429,8 @@ struct optee_msg_arg {
 #define OPTEE_MSG_RPC_SHM_TYPE_APPL	0
 /* Memory only shared with non-secure kernel */
 #define OPTEE_MSG_RPC_SHM_TYPE_KERNEL	1
+/* Memory, exported using dma-buf buffer sharing API */
```

Comment:
The OP-TEE message protocol doesn't care/know about dma-buf, please describe it with different words

Reply:
@jenswi-linaro I specified dma-buf just to explicitly differentiate from `OPTEE_MSG_RPC_CMD_SHM_ALLOC`, which also says that this memory is `shared with a non-secure user space application`.

I can specify that it's `/* Memory only shared with non-secure kernel, exported to userspace */` but people will be confused and don't understand what is the difference between `OPTEE_MSG_RPC_SHM_TYPE_APPL` (where tee-supplicant handles memory allocations) and `OPTEE_MSG_RPC_SHM_TYPE_DMABUF` (memory allocations are handled directly in optee module, without dispatching this request to NW userspace).

Reply:
This is some kind of global shared memory, that it happens to be implemented with dms-buf has nothing to do with it.